### PR TITLE
Secure storage maintenance

### PR DESCRIPTION
**Pull request by jenswi-linaro**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -144,26 +144,24 @@ script:
...
   - $make CFG_WITH_LPAE=y
-  - $make CFG_ENC_FS=n
-  - $make CFG_ENC_FS=y CFG_FS_BLOCK_CACHE=y
-  - $make CFG_ENC_FS=n CFG_FS_BLOCK_CACHE=y
+  - $make CFG_FS_BLOCK_CACHE=y
   - $make CFG_WITH_STATS=y
   - $make CFG_RPMB_FS=y
-  - $make CFG_RPMB_FS=y CFG_ENC_FS=n
-  - $make CFG_RPMB_FS=y CFG_ENC_FS=n CFG_RPMB_TESTKEY=y
+  - $make CFG_RPMB_FS=y
```
Comment (etienne-lms) [inline .travis.yml]: duplicated `$make CFG_RPMB_FS=y`

File: core/tee/tee_fs_rpc_cache.c [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+	size_t sz = size;
+	uint64_t c = 0;
+	paddr_t p;
+	void *va;
+
+	/*
+	 * Always allocate in page chunks as normal world allocates payload
+	 * memory as complete pages.
+	 */
+	if (!size)
```
Comment (etienne-lms) [inline core/tee/tee_fs_rpc_cache.c]: if request size is null, why allocating a small page. Could simply return NULL.
Yet such value is not used in current code.

File: core/tee/tee_rpmb_fs.c [hunk 3]
```diff
@@ -638,13 +625,11 @@ static TEE_Result tee_rpmb_req_pack(struct rpmb_req *req,
 			       RPMB_NONCE_SIZE);
 
 		if (rawdata->data) {
-#ifdef CFG_ENC_FS
 			if (fek)
 				encrypt_block(datafrm[i].data,
 					rawdata->data + (i * RPMB_DATA_SIZE),
 					*rawdata->blk_idx + i, fek);
 			else
```
Comment (etienne-lms) [inline core/tee/tee_rpmb_fs.c]: is it allowed here to not provide an encryption key?

Comment (jenswi-linaro) [inline core/tee/tee_rpmb_fs.c]: It's certainly used. If it's an issue or not is in my opinion not in scope of this P-R.

**Comment by jenswi-linaro** [comment]
Update

**Comment by jenswi-linaro** [comment]
The travis errors are likely to go away when squashing the review commits into their parent commits.

**Comment by jforissier** [comment]
@jenswi-linaro indeed. The combined diff is checked separately for convenience, see https://travis-ci.org/OP-TEE/optee_os/builds/166757803#L1769-L1772.

**Comment by etienne-lms** [comment]
tested on my b2260 with the GP tests extensions. no issues seen so far.
tested on qemu_v8: i found issues by these might be related to my setup. i rebase and test back.
I'm checking twice then i'll add a Tested-by tag.

**Comment by jenswi-linaro** [comment]
Rebased, tags applied

**Comment by jenswi-linaro** [comment]
Complaints from travis are false negatives from checkpatch.

**Comment by etienne-lms** [comment]
note: to properly test this, you need to change optee_test to get the 200xx tests enable.
These are conditioned to new deprecated `CFG_ENC_FS`.

**Comment by jenswi-linaro** [comment]
What's next step?

**Comment by jforissier** [comment]
I guess it's all good! Can you rebase, please? And you may add my T-b to the last commit:
`Tested-by: Jerome Forissier <jerome.forissier@linaro.org> (HiKey 32/64 GP)`
One last thing: you have a typo ('optimize') in the subject of the 3rd commit.

Thanks!

**Comment by jenswi-linaro** [comment]
Rebased and tag applied

**Pull request merged**